tests/kernel/wave/wave_e2e_test.py

```diff
@@ -17,15 +17,18 @@ def test_copy():
     N = tkl.sym.N
     BLOCK_M = tkl.sym.BLOCK_M
     BLOCK_N = tkl.sym.BLOCK_N
+    ELEMS_PER_THREAD = tkl.sym.ELEMS_PER_THREAD
     ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
 
     constraints: list[tkw.Constraint] = [
         tkw.HardwareConstraint(
-            threads_per_wave=64, waves_per_block=(1, 1, 1), vector_shapes={M: 16, N: 16}
+            threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
+            vector_shapes={M: 1, N: ELEMS_PER_THREAD},
         )
     ]
-    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 0)]
-    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 1)]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]

```

Comment:
Why are the workgroup ids being switched around?

Reply:
Discussed and resolved offline. 

For documentation purposes, we are switching since by convention we set `thread_per_block` to `(wave_count[0] * subgroup_size, wave_count[1], wave_count[2])`. So setting fastest dim to 0 allow us to distribute the fastest dim to the largest number of threads.

Reply:
Thanks, that makes sense. From the programmer's perspective this is not obvious, so I am in favor of not specifying the workgroup dim in the workgroup constraints and have the compiler assign the index to the innermost dimension (we can keep this as a power user argument).